exchanges/bitmex/bitmex_parameters.go

```diff
@@ -109,8 +109,8 @@ func (p APIKeyParams) ToURLVals(path string) (string, error) {
 }
 
 // IsNil checks to see if any values has been set for the paramater
-func (p APIKeyParams) IsNil() bool {
-	return p == (APIKeyParams{})
+func (p *APIKeyParams) IsNil() bool {
+	return p == (&APIKeyParams{})
```

Comment:
This is always going to eval false because you are checking for matching mem addr; possibly change to:
```go 
return *p == (APIKeyParams{})
```

Reply:
could simplify it more with return p == nil

Reply:
if you do a := APIKeyParams{}
then a.IsNIl() it will always be false as well if it compares nil.

Reply:
That's true! 💯